jsign-core/src/main/java/net/jsign/jca/AmazonSigningService.java

```diff
@@ -91,8 +91,8 @@ public class AmazonSigningService implements SigningService {
     public AmazonSigningService(String region, String credentials, Function<String, Certificate[]> certificateStore) {
         this.certificateStore = certificateStore;
 
-        // parse the credentials
-        String[] elements = credentials.split("\\|", 3);
+        // Obtain the credentials
+        String[] elements = credentials != null ? credentials.split("\\|", 3) : IMDS2Client.create().getCredentials();
         if (elements.length < 2) {
             throw new IllegalArgumentException("Invalid AWS credentials: " + credentials);
```

Comment:
Here if `credentials` was null and IMDS2Client didn't return the credentials, I think the IllegalArgumentException should reuse the error message removed from `KeyStoreType`

Reply:
Thank you for the changes. I'll move the credentials into a dedicated class, that will solve the propagation of `parameterName` to `AmazonSigningService`.

Reply:
So I went with a custom exception to flag that **AWS** metadata service is not reachable (which I should have done from the start...). Hopefully the error message provides some hint of what goes wrong, for a user to troubleshoot, but it means the message is not identical when running on GitHub Actions (Azure) or locally (I replaced the string comparison in the test with pattern matching).

Now, on Azure, we get:
```
jsign: storepass option must specify the AWS credentials: <accessKey>|<secretKey>[|<sessionToken>], when not running from an EC2 instance (IMDSv2 host did not respond as expected; are you in AWS cloud?)
```
and locally:
```
jsign: storepass option must specify the AWS credentials: <accessKey>|<secretKey>[|<sessionToken>], when not running from an EC2 instance (IMDSv2 host was unreachable; check the hop limit if containerized)
```